otherlibs/dune-rpc/private/versioned.mli

```diff
@@ -0,0 +1,92 @@
+(** The main logic for the runtime versioning protocol for the Dune RPC. For a
+    high-level explanation and rationale, see [doc/dev/rpc-versioning.ml]. *)
+
+open Stdune
+open Types
+
+module Menu : sig
+  type t
+
+  val default : t
+
+  (** For each method known by both local and remote, choose the highest common
+      version number. Returns [None] if the resulting menu would be empty. *)
+  val select_common :
+       local_versions:Int.Set.t String.Map.t
+    -> remote_versions:(string * int list) list
```

Comment:
I suggest we introduce abstract types for versions and method names. Otherwise it becomes hard to read the code and enforce invariants on these values.

Reply:
Are you thinking that we might want versions/method names to become more complex than strings/ints? I'm not opposed to introducing `Method_name`/`Method_version` newtypes, but it seems like adding a lot of boilerplate for not much gain (most of the invariants we'd want to enforce are external, like the no-duplicates rule, which is why `Menu` is abstracted).

Reply:
On second pass, you also mention that it becomes harder to read the code. I don't completely disagree, but  after a quick attempt to do this, in my opinion the code becomes more convoluted (with more constructors needing to be introduced and matched out) without much corresponding upside, as most of the time they are passed, uninspected, into the `Versioned` module.

Reply:
Okay, well they can at least be aliases in `Types`. That should give a readability boost and we can rethink making them abstract later.